internal/productcatalog/feature_connector.go

```diff
@@ -118,13 +117,13 @@ func (c *featureConnector) ListFeatures(ctx context.Context, params ListFeatures
 	return c.featureRepo.ListFeatures(ctx, params)
 }
 
-func (c *featureConnector) GetFeature(ctx context.Context, featureID models.NamespacedID) (Feature, error) {
-	feature, err := c.featureRepo.GetByID(ctx, featureID)
+func (c *featureConnector) GetFeature(ctx context.Context, namespace string, idOrKey string) (*Feature, error) {
+	feature, err := c.featureRepo.GetByIdOrKey(ctx, namespace, idOrKey, true)
```

Comment:
Please add an options struct at least for the includeArchived as true/false const arguments are extremely hard to understand what they are representing. (or you can add a type IncludeArchive bool so that the next one reading this code won't have to look up the function signature) ;)

Reply:
I agree 4 arguments is starting to get on the longer side, but signature wise a bool is no different than a string and we don't do custom types for that.